File: Pascal7601/reverse_int.py
"""
program that takes an integer as input and
and returns the integer with reversed digit
ordering
"""

def reverse(num: int) -> int:
  """
  function that returns a reversed integer
  """
  if num < 0: # store the sign of the number
    op = -1
  else:
    op = 1

  num_str = str(abs(num)) # store the absolute value of the number as sting
  reversed_num = int(num_str[::-1])

  return reversed_num * op

File: xander254/reverse_int.c
#include <stdio.h>

/**
  * reverse integer - A function that reverses the integer it input
  * @a: the integer to be reversed
  * Return: reversed integer
  */
int reverse_integer(int a)
{
        int revarr = 0;
        
        while (a != 0)
        {
            revarr = revarr *10 + (a % 10);
            a = a / 10;
        }   
        return (revarr);
}
int main()
{
    int num = 0;

    /*input*/
    printf("Enter an integer\n");
    scanf("%d", &num);


    /*reversed*/
    int result = reverse_integer(num);
    printf("The reversed value is: %d\n", result);

    return 0;
}
